mailer.http.class.php

```diff
@@ -291,7 +306,7 @@ protected function parse_reply_to()
             }
         }
 
-        return implode(',', $replyTos);
+        return apply_filters('wpsp_reply_to', implode(',', $replyTos));
```

Comment:
Hi @rajumsys, can you please help me with an example of how to use this filter (`wpsp_reply_to`) in my functions.php? I'm trying to set the replyto email address for all mail sent by this plugin. Thanks!

Reply:
You need to return an array for formatted replyTos. See https://github.com/SparkPost/wordpress-sparkpost/blob/master/mailer.http.class.php#L354-L358 for the formatting. 

Like

```
add_filter('wpsp_reply_to', function($replyTos) {
  //return formatted replytos. 
  return array_merge($replyTos, [
   'SomeName <someone@somedomain.com>'
 ]); 
  
}  
```

Reply:
@rajumsys thanks so much, I'm running this snippet in `functions.php`:

```php
add_filter('wpsp_reply_to', function($replyTos) {
    return array_merge($replyTos, [
        'Cloud Support <support@cloudchemistry.co>'
    ]);
});
```

But the `replyto` header is not being sent. Any tips? Thanks!

Reply:
Please add more context.

1) you are using http mailer
2) whether you're using a template. in that case, you'll need to add substitution tag in the template. 
3) try sending a test email and add whole debug info.

Reply:
@rajumsys 
1. I'm using the wordpress sparkpost plugin
2. I'm not using any templates
3. Test emails are getting sent successfully, but no debug info is being sent with them, even though I have the "Show email debugging messages" checkbox checked.

Reply:
@rajumsys my apologies, as regards number 3 above, I wasn't getting debug messages when I would send a test message from the live environment, but debug was getting printed if I send a test message from the dev environment. It's the quirks of [Pantheon](https://www.pantheon.io/) that I'm still getting used to.

Anyhoo, here's the debug output:

```bash
Connection: opening to smtp.sparkpostmail.com:2525, timeout=300, options=array ()
Connection: opened
SERVER -> CLIENT: 220 2.0.0 smtp.sparkpostmail.com ESMTP ecelerity 4.2.38.62433 r(:) Sat, 16 Dec 2017 15:13:46 +0000
CLIENT -> SERVER: EHLO appserver-09ee43b5.c.pantheon-dmz.internal
SERVER -> CLIENT: 250-b.mta1vsmtp.cc.aws-usw2b.prd.sparkpost says EHLO to 23.251.152.220:41236250-ENHANCEDSTATUSCODES250-STARTTLS250-PIPELINING250-AUTH=LOGIN PLAIN250-AUTH LOGIN PLAIN250 8BITMIME
CLIENT -> SERVER: STARTTLS
SERVER -> CLIENT: 220 2.0.0 continue
CLIENT -> SERVER: EHLO appserver-09ee43b5.c.pantheon-dmz.internal
SERVER -> CLIENT: 250-b.mta1vsmtp.cc.aws-usw2b.prd.sparkpost says EHLO to 23.251.152.220:41236250-AUTH=LOGIN PLAIN250-AUTH LOGIN PLAIN250-PIPELINING250-8BITMIME250 ENHANCEDSTATUSCODES
CLIENT -> SERVER: AUTH LOGIN
SERVER -> CLIENT: 334 VXNlcm5hbWU6
CLIENT -> SERVER: U01UUF9JbmplY3Rpb24=
SERVER -> CLIENT: 334 UGFzc3dvcmQ6
CLIENT -> SERVER: MGQyM2M3M2E2OWNmNDNlZDE3ZWVkZmUxNjM5ZWM0Njg2OTNhNzdlOQ==
SERVER -> CLIENT: 235 2.0.0 Authed. Go on.
CLIENT -> SERVER: MAIL FROM:<james@cloudchemistry.email>
SERVER -> CLIENT: 250 2.0.0 MAIL FROM accepted
CLIENT -> SERVER: RCPT TO:<jamesfinn@techie.com>
SERVER -> CLIENT: 250 2.0.0 RCPT TO accepted
CLIENT -> SERVER: DATA
SERVER -> CLIENT: 354 3.0.0 continue. finished with "\r\n.\r\n"
CLIENT -> SERVER: Date: Sat, 16 Dec 2017 15:13:45 +0000
CLIENT -> SERVER: To: jamesfinn@techie.com
CLIENT -> SERVER: From: James from Cloud Chemistry <james@cloudchemistry.email>
CLIENT -> SERVER: Subject: SparkPost email test
CLIENT -> SERVER: Message-ID: <71361f59c48247423b29c3367c267ab3@appserver-09ee43b5.c.pantheon-dmz.internal>
CLIENT -> SERVER: X-Mailer: wordpress-sparkpost/3.0.1 on PHPMailer 5.2.22 (https://github.com/PHPMailer/PHPMailer)
CLIENT -> SERVER: X-MSYS-API: {"options":{"open_tracking":false,"click_tracking":false,"transactional":true,"sandbox":false}}
CLIENT -> SERVER: MIME-Version: 1.0
CLIENT -> SERVER: Content-Type: multipart/mixed;
CLIENT -> SERVER: boundary="b1_71361f59c48247423b29c3367c267ab3"
CLIENT -> SERVER: Content-Transfer-Encoding: 8bit
CLIENT -> SERVER: 
CLIENT -> SERVER: This is a multi-part message in MIME format.
CLIENT -> SERVER: 
CLIENT -> SERVER: --b1_71361f59c48247423b29c3367c267ab3
CLIENT -> SERVER: Content-Type: text/html; charset=us-ascii
CLIENT -> SERVER: 
CLIENT -> SERVER: <h3>Hurray!!</h3><p>You've got mail! <br/><br> Regards,<br/><a href="https://www.sparkpost.com">SparkPost</a> WordPress plugin</p>
CLIENT -> SERVER: 
CLIENT -> SERVER: 
CLIENT -> SERVER: --b1_71361f59c48247423b29c3367c267ab3
CLIENT -> SERVER: Content-Type: text/plain; name="sample.txt"
CLIENT -> SERVER: Content-Transfer-Encoding: base64
CLIENT -> SERVER: Content-Disposition: attachment; filename=sample.txt
CLIENT -> SERVER: 
CLIENT -> SERVER: VGhpcyBpcyBhIHNhbXBsZSBmaWxlIHRvIGJlIHNlbnQgYXMgYW4gYXR0YWNobWVudC4g
CLIENT -> SERVER: 
CLIENT -> SERVER: 
CLIENT -> SERVER: --b1_71361f59c48247423b29c3367c267ab3--
CLIENT -> SERVER: 
CLIENT -> SERVER: .
SERVER -> CLIENT: 250 2.0.0 OK CF/B6-28077-A28353A5
CLIENT -> SERVER: QUIT
SERVER -> CLIENT: 221 2.3.0 b.mta1vsmtp.cc.aws-usw2b.prd.sparkpost closing connection
Connection: closed
```

And, thanks again for your assistance!

Reply:
So, you're using SMTP! Which is fine but HTTP API would be easier to debug. I'll check a few things myself first. 

However, you need to delete your API Key from sparkpost account as this log contains your (encoded but easy to decode) API Key. To avoid problem, please create and use a new key.

Reply:
@JamesJosephFinn 
So, turns out, I was slightly wrong. A few things

1) This hook is supported only for HTTP API. So, it won't work as you're using SMTP. Please change to HTTP API (unless you've any issues with that). Please add appropriate permissions to the API key to use with HTTP)
2) the argument passed in the filter callback function isn't an array. So `array_merge` won't work. My following code worked
```
add_filter('wpsp_reply_to', function($replyTos) {
    return 'Cloud Support <support@cloudchemistry.co>';
});
```
This is ignoring existing reply to addresses passed in `$replyTos`. Please handle them as you need. 


*Important:* Remove the old API key (see my previous comment).

Reply:
In case you need to use reply to with SMTP, please feel free to create an issue (or even better a PR :) ).

Reply:
@rajumsys thank you, this clears everything up, and I've reset the API key, thanks for pointing that out, greatly appreciated.

Reply:
I had actually set the method as `SMTP` for no explicit reason, and had forgotten that I had done so. Thanks again for your assistance. Very much appreciated.

Reply:
Sure, so things are fine now?

Reply:
@rajumsys yes, indeed, sir. Thanks again for your help!